lib/utils/utils.js

```diff
@@ -270,3 +275,62 @@ function findUnorderedProperty(arr) {
 
   return null;
 }
+
+/**
+ * Checks whether a computed property has duplicate dependent keys.
+ *
+ * @param  {CallExpression} callExp Given call expression
+ * @return {Boolean}        Flag whether dependent keys present.
+ */
+function hasDuplicateDependentKeys(callExp) {
+  if (!ember.isComputedProp(callExp)) return false;
+
+  const dependentKeys = parseDependentKeys(callExp);
+  const uniqueKeys = dependentKeys
+    .filter((val, index, self) => self.indexOf(val) === index);
+  return uniqueKeys.length !== dependentKeys.length;
+}
+
+/**
+ * Parses dependent keys from call expression and returns them in an array.
+ *
+ * It also unwraps the expressions, so that `model.{foo,bar}` becomes `model.foo, model.bar`.
+ *
+ * @param  {CallExpression} callExp CallExpression to examine
+ * @return {Array}          Array of unwrapped dependent keys
+ */
+function parseDependentKeys(callExp) {
+  const isMemberExpCallExp = callExp.arguments.length === 0 && callExp.callee.object;
+  const args = isMemberExpCallExp ? callExp.callee.object.arguments : callExp.arguments;
+
+  const dependentKeys = args
+    .filter(arg => isLiteral(arg))
+    .map(literal => literal.value);
+  return unwrapBraceExpressions(dependentKeys);
+}
+
+/**
+ * Unwraps brace expressions.
+ *
+ * @param  {Array} dependentKeys array of strings containing unprocessed dependent keys.
+ * @return {Array} Array of unwrapped dependent keys
+ */
+function unwrapBraceExpressions(dependentKeys) {
+  const braceExpressionRegexp = /\.{.+}/g;
+
+  const unwrappedExpressions = dependentKeys.map((key) => {
+    if (!braceExpressionRegexp.test(key)) return key;
+
+    const parts = key.split('.');
+    const prefix = parts[0];
+    const properties = parts[1]
+      .replace('{', '')
+      .replace('}', '')
+      .split(',');
+
+    return properties.map(property => `${prefix}.${property}`);
+  });
+
+  return unwrappedExpressions
```

Comment:
on the other hand, this will not work in Ember, as only one level of nesting is supported, WDYT @michalsnik?